src/core/reactor.cc

```diff
@@ -1921,6 +1924,114 @@ reactor::inotify_add_watch(int fd, std::string_view path, uint32_t flags) {
     });
 }
 
+future<std::tuple<pid_t, int, int>>
+reactor::spawn(std::string_view pathname,
+               const std::vector<std::string_view>& argv,
```

Comment:
You copy argv and env inside do_with anyway, why not making the API function accept argv/env vectors by value from the very beginning?

Reply:
okay. changed to 
```c++
  reactor::spawn(std::string_view pathname,
                 std::vector<sstring> argv,
                 std::vector<sstring> env)
```